manifests/agent/service/systemd.pp

```diff
@@ -4,6 +4,7 @@
   Boolean                 $enabled                             = false,
   Optional[Integer[0,23]] $hour                                = undef,
   Variant[Integer[0,59], Array[Integer[0,59]], Undef] $minute  = undef,
+  Optional[String] $timezone                                   = undef,
```

Comment:
are emtpy strings allowed, or can we narrow down the datatype a bit?

Reply:
it works as expected when not providing that param.
It does change the timer config as a space is then added, but the timer still works.

Reply:
`(/Stage[main]/Puppet::Agent::Service::Systemd/Systemd::Timer[puppet-run.timer]/Systemd::Unit_file[puppet-run.timer]/File[/etc/systemd/system/puppet-run.timer]/content) -OnCalendar=*-*-* *:24,54:00
(/Stage[main]/Puppet::Agent::Service::Systemd/Systemd::Timer[puppet-run.timer]/Systemd::Unit_file[puppet-run.timer]/File[/etc/systemd/system/puppet-run.timer]/content) +OnCalendar=*-*-* 10:0:00 Europe/Brussels

result in systemctl show puppet-run.timer
TimersCalendar={ OnCalendar=*-*-* 10:00:00 Europe/Brussels ; next_elapse=Sat 2025-04-26 08:00:00 UTC }`

Reply:
yes, but do we want to allow `''` as values? I think that doesn't make sense and it should be `String[1]`, so empty strings aren't allowed?

Reply:
I can add that, but having an empty, or multiple spaces even, string would still be ok, as this just gets added to the end of the line in the systemd timer config file, so that will not make any difference at all.